src/components/ExperienceTimeline.tsx

```diff
@@ -1,23 +1,77 @@
 import { Box, Text } from "@chakra-ui/react";
-import { ExperienceDetail } from "./Experience";
-import styles from "./ExperienceTimeline.module.css";
+import { Experience } from "../consts";
 
 interface ExperienceTimelineProps {
-  experienceDetails: ExperienceDetail[];
-  updateExperienceDataIndex: (index: number) => void;
+  experienceDetails: Experience[];
+  setExperience: (experience: Experience) => void;
 }
 
+const getPseudoElementStyles = (isHovered: boolean) => {
+  let width = "0.313rem";
+  let height = "0.313rem";
+  let top = "0.625rem";
+  let left = "-0.05rem";
+  let boxShadow =
+    "0 0 0 4px grey, inset 0 2px 0 rgb(0 0 0 / 8%), 0 3px 0 4px rgb(0 0 0 / 5%)";
+  let background = "black";
+
+  if (isHovered) {
+    width = "0.5rem";
+    height = "0.5rem";
+    top = "0.75rem";
+    left = "-0.15rem";
+    boxShadow =
+      "0 0 0 4px black, inset 0 2px 0 rgb(0 0 0 / 8%), 0 3px 0 4px rgb(0 0 0 / 5%)";
+    background = "white";
+  }
+
+  return {
+    content: `""`,
+    position: "absolute",
+    width,
+    height,
+    top,
+    left,
+    borderRadius: "50%",
+    zIndex: "1",
+    boxShadow,
+    background,
+    transition: "all 0.3s ease-in-out",
+    cursor: "pointer",
+  };
+};
+
 const ExperienceTimeline = ({
   experienceDetails,
-  updateExperienceDataIndex,
+  setExperience,
 }: ExperienceTimelineProps) => {
   return (
-    <Box className={styles.timeline}>
-      {experienceDetails.map((experience, index) => (
+    <Box
+      position="relative"
+      _after={{
+        content: `""`,
+        position: "absolute",
+        width: "0.2rem",
+        background: "black",
+        top: 0,
+        bottom: 0,
+      }}
+    >
+      {experienceDetails.map((experience) => (
         <Box
-          className={styles.experienceSummary}
-          key={experience.role + index}
-          onClick={() => updateExperienceDataIndex(index)}
+          key={experience.toString()}
```

Comment:
We should not use entire experience objects' serialized value as key.  We can assign a separate ID to each experience and use it as a key. We can also use the index as a key as I find it difficult to think of a case where we will change the orientation of the experience list once it has been finalized.